src/marshmallow/fields.py

```diff
@@ -634,9 +640,14 @@ def _serialize(self, nested_obj, attr, obj, **kwargs):
         return schema.dump(nested_obj, many=many)
 
     def _test_collection(self, value):
+        if self.many and not self.schema.many:
+            raise FieldConfigurationError(
+                '"many" may not be used in conjunction with an instance in a nested field.'
+            )
```

Comment:
I'd remove this test. This should be coverered by the init warning you added above, right?